### PR TITLE
Add a contribution guide

### PR DESCRIPTION
**Pull request by PeterJCLaw**
This records the current expectations of contributions as well as the direction the runbook is going. Like many things in the runbook this is probably incomplete, however it drastically improves the current state (i.e: not having any explicit guidance).

Fixes https://github.com/srobo/runbook/issues/1

### CODE REVIEW

File: docs/contributing.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
+# Contributing
```
Comment (RealOrangeOne) [inline docs/contributing.md:1]: I wonder if this content should actually be in the `CONTRIBUTING.md` file, and then embedded in the live site via the markdown embedding thing? That way the contributing doc is nice and reasonably self-contained, but the site still also contains the important information.

Comment (PeterJCLaw) [reply docs/contributing.md:1]: Yeah, I'd tried that initially. Unfortunately getting the links to play nicely (without resorting to linking out to the hosted version) was rather complicated.

File: docs/contributing.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+volunteers, regardless of their situation. As a result, the runbook is organised
+into topics which relate to volunteers as a whole, rather than any specific
+group.
+
+This is a deliberate change from a previous approach where content was arranged
+around the people who worked on specific areas. Unfortunately that approach lead
+to isolated silos of information which varied wildly in their internal
+structure, content approach, utility and discoverablity. While the runbook was
+pulled together from a number of those sources, it
+
```
Comment (sersorrel) [inline docs/contributing.md:60]: it

File: docs/contributing.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+muscle-memory to many contributors and readers. Making big changes to it
+therefore needs careful consideration to ensure that their workflows are not
+disrupted unduly.
+
+It's a good idea therefore to raise any large changes as a [new issue][new-issue]
+before commiting to them.
+
+Alternatively, if your change is more easily explained as a demonstration of the
+outcome, it's fine to raise a pull request showing that. Note however that your
+suggestion will still be discsused and may not be accepted, so you may wish to
```
Comment (trickeydan) [inline docs/contributing.md:26]: I think that we should await the resolution of ownership and control of the runbook before finalising this section. Please see the email sent on 14/02

File: docs/contributing.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+For any processes, try to ensure there are no dead ends.
+
+For example: if we need to contact someone about something, ensure the person
+being contacted will have the ability to resolve the situation rather than
+themselves needing to contact someone else. (Note that contact-by-proxy is ok,
+as long as the eventual recipient meets the preceding criteria)
+
+This does _not_ mean that the runbook needs to exhaustively document all
+possible outcomes! For examepl: it may be the case that someone needs to use
```
Comment (trickeydan) [inline docs/contributing.md:92]:
```suggestion
possible outcomes! For example: it may be the case that someone needs to use
```

**Review by trickeydan — COMMENTED** [review]
Thanks for writing this. It will be a great resource when the runbook comes into wider use.

We should consider adding a spellchecker (mdspell?) as there are a number of typos in this PR.

I kindly request that you hold off merging this PR until the "runbook situation" has been resolved, see email sent on 14/02. I request this because the content in this PR asserts some things that may not necessarily be the case after that discussion and we should avoid publishing conflicting information where possible.

File: docs/contributing.md (hunk 2 above)
Comment (PeterJCLaw) [reply docs/contributing.md]: oops, though I'd sorted that before pushing :( 41aa90b927cd4d7d109a3b5ebe4f1a53f6d87463 finishes the sentence, though I'm still not entirely happy with the wording.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# Contributing
+
+Contributions to Student Robitics' Runbook should:
```
[the remark this answers is not in the document]
Comment (PeterJCLaw) [reply CONTRIBUTING.md]: 939a6a6827e4666e3f8ae96e2d6df76d93fe26aa

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+* follow Student Robitics' [Code of Conduct][code-of conduct]
```
[the remark this answers is not in the document]
Comment (PeterJCLaw) [reply CONTRIBUTING.md]: 939a6a6827e4666e3f8ae96e2d6df76d93fe26aa

File: docs/contributing.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
+# Contributing
+
+Contributions to Student Robitics' Runbook should follow Student Robitics'
```
[the remark this answers is not in the document]
Comment (PeterJCLaw) [reply docs/contributing.md]: 29bce0d149f5378f4cf77e84e2a67becb1d15b1d

File: docs/contributing.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (PeterJCLaw) [reply docs/contributing.md]: 29bce0d149f5378f4cf77e84e2a67becb1d15b1d

Comment (PeterJCLaw) [reply docs/contributing.md]: This PR aims to record the current state -- i.e: the original vision for the runbook.

File: docs/contributing.md (hunk 4 above)
Comment (PeterJCLaw) [reply docs/contributing.md]: 29bce0d149f5378f4cf77e84e2a67becb1d15b1d

**Review by PeterJCLaw — COMMENTED** [review]
Thanks for your input and for catching those silly typos.

> We should consider adding a spellchecker (mdspell?) as there are a number of typos in this PR. (There is an issue #10 for this)

Yeah, there's a ticket for that (https://github.com/srobo/runbook/issues/10), though I agree it's increasingly annoying not to have it in place.
I also need to work out why my primary dev. environment isn't picking these up :(

> I kindly request that you hold off merging this PR until the "runbook situation" has been resolved, see email sent on 14/02. I request this because the content in this PR asserts some things that may not necessarily be the case after that discussion and we should avoid publishing conflicting information where possible.

This PR in general aims to record the original vision for the runbook in combination with common patterns for working with it which were established during its initial formation.
Unfortunately, if anything, it is likely that there are still things which are missing from this -- it's often hard to know things you already know.
i realise that this was missing previously (and its omissions may have lead to some of the prior conflict), however I don't believe that that changes its validity.

**Review by RealOrangeOne — APPROVED** [review]

**Pull request merged**